File: team2/ruby/rasa.rb
#!/usr/bin/env ruby

tl = 0
tc = 0
ARGV.each do|a|
  content = ''
  File.open(a).each do |line|
    content += line
  end
  lines = 0
  complexity = 0
  depth = 1
  content.split("").each do |i|
    if i == "{"
      depth = depth + 1
    end
    if i == "}"
      depth = depth - 1
    end
    if i == ";"
      complexity = complexity + depth
      tc = tc + depth
    end
    if i == "\n"
      lines = lines + 1
      tl = tl + 1
    end
  end
  puts "file #{a}: lines: #{lines} RaSa: #{complexity}"
end
puts "total lines: #{tl} RaSa: #{tc}"

File: team2/c++/RaSa.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

void processFile(int& tl, int& tc, const char* path) {
	string line;
	ifstream infile(path);
	int depth = 1;
	int lines = 0; int complexity = 0;
	while (getline(infile, line))
	{
		for (int i = 0; i < line.length(); ++i) {
			if (line[i] == '{') {
				depth++;
			}
			if (line[i] == '}') {
				depth--;
			}
			if (line[i] == ';') {
				complexity += depth;
			}
		}
		lines++;
	}
	cout << "file: " << path << " lines: " << lines << " RaSa: " << complexity << endl;
	tl += lines; tc += complexity;
}

int main(int argc, char** argv) {
	int tl = 0; int tc = 0;
	for (int i = 1; i < argc; ++i) {
		processFile(tl, tc, *(argv + i));
	}
	cout << "total lines: " << tl << " RaSa: " << tc << endl;
}
